src/Tribe/Event_Repository.php

```diff
@@ -267,4 +312,162 @@ public function filter_by_has_rsvp( $has_rsvp = true ) {
 		$this->decorated->where_clause( "{$prefix}_ticket_event.meta_id IS NULL 
 			OR {$prefix}_ticket_event.meta_key != '_tribe_rsvp_for_event'" );
 	}
+
+	/**
+	 * Filters events to include only those that match the provided attendee(s).
+	 *
+	 * @since TBD
+	 *
+	 * @param int|array $attendee_ids The attendee(s) to filter by.
+	 */
+	public function filter_by_attendee( $attendee_ids ) {
+		global $wpdb;
+
+		$attendee_ids = (array) $attendee_ids;
+		$attendee_ids = array_map( 'absint', $attendee_ids );
+		$attendee_ids = array_unique( $attendee_ids );
+		$attendee_ids = implode( ', ', $attendee_ids );
+
+		$alias_event = 'ticket_attendee_event';
+
+		$event_meta_keys = $this->attendee_to_event_keys();
+		$event_meta_keys = array_map( [ $wpdb, '_real_escape' ], $event_meta_keys );
+		$event_meta_keys = '"' . implode( '", "', $event_meta_keys ) . '"';
+
+		// Join to the meta that relates attendees to events.
+		$this->decorated->filter_query->join( "
```

Comment:
:no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: 0d352cda053c7ba7cf8301f4de59efab</sub>